util/argo/argo.go

```diff
@@ -115,6 +115,43 @@ func FilterByProjects(apps []argoappv1.Application, projects []string) []argoapp
 	return items
 }
 
+func FilterByPath(apps []argoappv1.Application, path string) []argoappv1.Application {
+	filteredApps := make([]argoappv1.Application, 0)
+	for _, app := range apps {
+		if app.Spec.Source != nil && (app.Spec.Source.Path == path || strings.HasPrefix(app.Spec.Source.Path, path+"/")) {
```

Comment:
```suggestion
		if app.Spec.Source != nil && (app.Spec.Source.Path == path || strings.HasPrefix(app.Spec.Source.Path, path+"/")) {
```
Checking for a path is simply not sufficient. You should also check for a absolute/relative path too. For eg. in any  setup, I should be simply able to pass the file path to a file that exists locally and it should internally check if any application is using that specified path for sync operation.  

@agaudreault Isn't this what we plan to do? Otherwise with the current setup we just pass an argument to the path flag and check if that matches with the source path.